### ISSUE TITLE
adb logcat can't  export

### ISSUE THREAD

**Issue by 0x88l** [openstf/stf#506]
I find some phones even  'nexus 5x' ,'huawei honor 6' , 'lenovo' can't export logcat.
I have used cmd 'adb logcat -B' to export logcat.
Compared the normal devices ,I found something wrong with the err devices.
The err devices  output the first line have some garbled like this.
![image](https://cloud.githubusercontent.com/assets/18346467/21413905/c0716f00-c835-11e6-8239-7bf996ab03d6.png)

It that caused the logcat can't export ?

**Comment by sorccu** [comment]
Are you using STF 2.1.0 or newer?

**Comment by 0x88l** [comment]
Yes

**Comment by sorccu** [comment]
Ok can you do a logcat dump and attach it to this issue? I think the options are `adb logcat -B -d >out.log`.

**Comment by 0x88l** [comment]
You means to upload the logcat file?

**Comment by sorccu** [comment]
Yeah just dump the raw output with that command and attach the the to this issue. Or send it to simo @ {name of this github org} .io if you don't want to post it here.

**Comment by sorccu** [comment]
That file parses fine with [adbkit-logcat](https://github.com/openstf/adbkit-logcat). If you're using the latest STF, is it possible you may have forgot to run `npm install` to update the dependencies?

**Comment by 0x88l** [comment]
I have used to adb-kit to write a test code to locate the problem.But adb.openlogcat can't read this nexus phone' logcat.It doesn't work too.Maybe something special like this  kinds of phone.
The code :
var adbkit = require('adbkit')
//var Promise = require('bluebird')
var adb = adbkit.createClient({host:'127.0.0.1', port:5037})
adb.Keycode = adbkit.Keycode
var serial = 'D8YDU16115003258'
//var serial='012e8e2764d6bccc'

adb.openLogcat(serial,{
      clear: true
    })
      .then(function (logcat) {
        function entryListener(entry){
          console.log(entry)   //      nexus  entry nothing
        }
        logcat.on('entry', entryListener)
        logcat.on('error',function (err) {
         console.log('entry error')
         console.error(err)

        })
    })

**Comment by 0x88l** [comment]
@sorccu 
I have found the problem.Because the phone does not have command(logcat -c) permissions.Do  you have a better idea to solve it?
Like this:
![image](https://cloud.githubusercontent.com/assets/18346467/21474188/b8737db0-cb56-11e6-9c93-0511f1cc2acc.png)
